packages/widget/src/utils.ts

```diff
@@ -24,3 +27,57 @@ const groupWhileHelper = <T>(
   }
   return groupWhileHelper([...currentGroup, head], tail, breakGroup);
 };
+
+// https://stackoverflow.com/a/55243880/11698825
+
+export class ReactHandler<PropsType> {
```

Comment:
I think it's a very useful finding from my side, we could use it in other places too